### PR TITLE
feat(ai): AI增加多机多卡 分布式训练

### PR DESCRIPTION
**Pull request by ZihanChen821**
### 改动
1. 集群信息新增gpuType和vramMb 参数，同时训练时需要将gpuType传回给适配器
2. 新增多机多卡功能，当用户在AI 训练时，如果选择多个节点（pod），则需要指定对应的算法框架，目前支持 tensorflow、pytorch和mindspore（华为特有），来进行对应的分布式训练
image

3. AI应用配置文件新增tag,并且新增三个接口：listApp, listTags,以及根据appId 来获取对应可创建该应用的集群信息 listClusters，这部分接口主要是为了后续 AI作业模块重构准备。

4. gpu分区提交作业和应用必传gpuType告知适配器是是什么gpu类型

5. 若是华为gpu卡，不管是不是分布式训练，提交应用和作业都需要指定框架

### CODE REVIEW

File: libs/protos/scheduler-adapter/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "build/index.js",
   "private": true,
   "scripts": {
-    "generate": "rimraf generated && buf generate --template buf.gen.yaml https://github.com/PKUHPC/scow-scheduler-adapter-interface.git#branch=ai-multi-pod-train",
+    "generate": "rimraf generated && buf generate --template buf.gen.yaml https://github.com/PKUHPC/scow-scheduler-adapter-interface.git#branch=feat-ai-release",
```
Comment (ddadaal) [inline libs/protos/scheduler-adapter/package.json:8]: 这个功能准备好了吗？如果准备好了，适配器接口需要发布一个新的版本，并且这个功能需要检查集群的适配器的版本是否是已经实现了新的接口的新版本。

**Review by ddadaal — CHANGES_REQUESTED** [review]

File: libs/protos/scheduler-adapter/package.json (hunk 1 above)
Comment (OYX-1) [reply libs/protos/scheduler-adapter/package.json:8]: 现在scow的master上就是feat-ai-release这个分支，不是版本号。
要在adapter-interface仓库里把feat-ai-release合到master中才能继续之前的检查适配器版本操作。
当时好像是说ai模块还是beta版就先不合

Comment (ddadaal) [reply libs/protos/scheduler-adapter/package.json:8]: https://github.com/PKUHPC/SCOW/pull/1361 已经合并了，这个PR的proto修改是只对AI的吧？那就

1. 在接口项目中，发起一个PR，把这个功能所用到的修改合并到ai的分支里
2. 当接口合并到AI的分支后，SCOW项目的本分支合并master，然后在ai项目用的proto项目里用ai的分支

**Review by OYX-1 — APPROVED** [review]

**Review by ddadaal — CHANGES_REQUESTED** [review]
https://github.com/PKUHPC/scow-scheduler-adapter-interface/pull/25#discussion_r1686618467

**Review by ddadaal — APPROVED** [review]
